torch/testing/_internal/common_utils.py

```diff
@@ -1309,30 +1336,148 @@ def check_if_enable(test: unittest.TestCase):
         if not getattr(test, test._testMethodName).__dict__.get('slow_test', False):
             raise unittest.SkipTest("test is fast; we disabled it with PYTORCH_TEST_SKIP_FAST")
 
-# Acquires the comparison dtype, required since isclose
-# requires both inputs have the same dtype, and isclose is not supported
-# for some device x dtype combinations.
-# NOTE: Remaps bfloat16 to float32 since neither the CPU or CUDA device types
-#  support needed bfloat16 comparison methods.
-# NOTE: Remaps float16 to float32 on CPU since the CPU device type doesn't
-#   support needed float16 comparison methods.
-# TODO: Update this once bfloat16 and float16 are better supported.
-def get_comparison_dtype(a, b):
-    # TODO: update this when promote_types supports bfloat16 and/or
-    # isclose supports bfloat16.
-    a_dtype = torch.float32 if a.dtype is torch.bfloat16 else a.dtype
-    b_dtype = torch.float32 if b.dtype is torch.bfloat16 else b.dtype
-
-    compare_dtype = torch.promote_types(a_dtype, b_dtype)
-
-    # non-CUDA (CPU, for example) float16 -> float32
-    # TODO: update this when isclose is implemented for CPU float16
-    if (compare_dtype is torch.float16 and
-        (a.device != b.device or a.device.type != 'cuda' or
-            b.device.type != 'cuda')):
-        compare_dtype = torch.float32
-
-    return compare_dtype
+
+class RelaxedBooleanPair(BooleanPair):
```

Comment:
Would you elaborate on this?

How bad would not supporting this be?

Reply:
Also this should probably add a note about the relationship between these pairs, assertEqual, and assertClose so future maintainers (including ourselves!) understand why these pairs are here and how the best way to modify in the future

Reply:
> Would you elaborate on this?

The `BooleanPair` added in #67794 only handles `bool`'s and `np.bool_`'s. In our test suite there are many tests that compare `bool`'s against 0d tensors with dtype `torch.bool`. In some cases even something like `False` vs `torch.tensor(0, dtype=torch.int64)`.

> How bad would not supporting this be?

We would need to manually fix every single test that does something like I explained above. Worse, if we try to get rid of all `Relaxed*Pair`'s this PR will grow unreasonable large since we will need to touch a good portion of our test suite. Have a look at  #58981 where I had some automation in `assertEqual`, have already "fixed" a lot of tests, and still are not close to being done. 

Having the ability to use different pairs for our internal test suite was the major reason to refactor `assert_close` in the first place. If we want more strictness in the future, I suggest we add the relaxed pairs for now and phase them out later one by one.